# Um panorama educacional da maior universidade do Reino Unido

## 1. Business Understanding

### 1.1. Contexto

A Open University é uma universidade pública britânica que possui o maior número de alunos de graduação no Reino Unido. Este conjunto de dados pertence à Plataforma de Aprendizagem Online da Open University com 7 arquivos CSV que contemplam 7 cursos ofertados nos anos de 2013 e 2014.

Esta é uma ótima oportunidade de avaliar a educação por diferentes prismas educacionais e sociais dada a robustez do conjunto de dados ofertados. O dataset também oferece a possibilidade de avaliar habilidades de tomada de decisão, visão analítica, limpeza e análise de dados e, por fim, não menos importante, o direcionamento ao negócio.

Dataset:
https://www.kaggle.com/datasets/anlgrbz/student-demographics-online-education-dataoulad/data

Felipe Rodrigues
04/2026

### 1.2. Objetivos

- Realizar a limpeza e normalização dos dados;
- Promover a análise exploratória dos dados (EDA) da Open University;
- Gerar insights sobre a operação educativa da Open University;
- Propor ações de melhorias com base nos dados.

### 1.3. Perguntas Iniciais

- Como fatores socio-economicos impactam a performance?
- Qual é o padrão de engajamento dos alunos?
- Quais cursos apresentam maior desempenho e quais os de maior abandono?
- No curso de maior evasão, é possível identificar o ponto de gargalo?
- Existe correlação entre alunos que revisam mais o conteúdo com alunos de maior performance?

### 1.4. Data Description

![Texto Alternativo](../Images/schema_oulad.png)

Abaixo estão as descrições de cada coluna para os 7 datasets que compõem o ecossistema **OULAD**.

#### **assessments.csv**
| Coluna | Descrição |
| :--- | :--- |
| `code_module` | ID do curso (identificador). |
| `code_presentation` | ID para turma, composto de ANO + PERÍODO (ex: "2013B" para Fevereiro, "2013J" para Julho). |
| `id_assessment` | ID da avaliação. |
| `assessment_type` | Tipo de avaliação: Tutor Marked Assessment (TMA), Computer Marked Assessment (CMA) ou Final Exam. |
| `date` | Data final para submissão (número de dias desde o início do curso). |
| `weight` | Peso da avaliação em %. Exames geralmente valem 100%; já a soma das outras avaliações é 100%. |

#### **courses.csv**
| Coluna | Descrição |
| :--- | :--- |
| `code_module` | ID do curso (identificador). |
| `code_presentation` | ID para turma, composto de ANO + PERÍODO (ex: "2013B" para Fevereiro, "2013J" para Julho). |
| `module_presentation_length` | Duração do curso em dias. |

#### **studentAssessment.csv**
| Coluna | Descrição |
| :--- | :--- |
| `id_assessment` | ID da avaliação. |
| `id_student` | ID único do estudante. |
| `date_submitted` | Data de submissão da avaliação (número de dias desde o início do curso). |
| `is_banked` | Flag indicando se o resultado da avaliação foi transferido de um curso anterior. |
| `score` | Nota do estudante nesta avaliação (0-100). Notas abaixo de 40 são consideradas Reprovação (Fail). |

#### **studentInfo.csv**
| Coluna | Descrição |
| :--- | :--- |
| `code_module` | ID do curso (identificador). |
| `code_presentation` | ID para turma, composto de ANO + PERÍODO (ex: "2013B" para Fevereiro, "2013J" para Julho). |
| `id_student` | ID único do estudante. |
| `gender` | Gênero do estudante. |
| `region` | Região geográfica onde o estudante vivia durante o curso. |
| `highest_education` | Nível de escolaridade mais alto ao ingressar. |
| `imd_band` | Índice de Privação Múltipla que categoriza a população em decis. É uma estatística oficial que mede a pobreza de uma área. |
| `age_band` | Faixa etária do estudante. |
| `num_of_prev_attempts` | Número de vezes que o estudante tentou este curso anteriormente. |
| `studied_credits` | Número total de créditos dos cursos que o estudante está cursando atualmente. |

#### **studentRegistration.csv**
| Coluna | Descrição |
| :--- | :--- |
| `code_module` | ID do curso (identificador). |
| `code_presentation` | ID para turma, composto de ANO + PERÍODO (ex: "2013B" para Fevereiro, "2013J" para Julho). |
| `id_student` | ID único para o estudante. |
| `date_registration` | Data do registro do estudante no curso (relativo ao início do curso). |
| `date_unregistration` | Data do cancelamento do registro (evasão), relativa ao início do curso. |

#### **studentVle.csv**
| Coluna | Descrição |
| :--- | :--- |
| `code_module` | ID do curso (identificador). |
| `code_presentation` | ID para turma, composto de ANO + PERÍODO (ex: "2013B" para Fevereiro, "2013J" para Julho). |
| `id_student` | ID único para o estudante. |
| `id_site` | ID do material do VLE (Ambiente Virtual de Aprendizagem). |
| `date` | Data da interação do estudante com o material (número de dias desde o início). |
| `sum_click` | Número de vezes que o estudante interagiu com o material naquele dia. |

#### **vle.csv**
| Coluna | Descrição |
| :--- | :--- |
| `id_site` | D do material do VLE. |
| `code_module` | ID do curso (identificador). |
| `code_presentation` | ID para turma, composto de ANO + PERÍODO (ex: "2013B" para Fevereiro, "2013J" para Julho). |
| `activity_type` | Tipo de atividade associada ao material do curso. |
| `week_from` | Semana a partir da qual o material está planejado para ser usado. |
| `week_to` | Semana até a qual o material está planejado para ser usado. |

## 2. Data Understanding

### 2.1. Setup (Library | Data)

#### 2.1.1. Import de Bibliotecas

In [1]:
import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt
import seaborn as sns

print('Bibliotecas importadas com sucesso.')

Bibliotecas importadas com sucesso.


#### 2.1.2. Read

Verificação da pasta com arquivos e atribuição dos CSVs em DataFrames

In [2]:
try:
    arquivos = os.listdir('../Dataset')

except FileNotFoundError:
    print('A pasta Dataset não foi encontrada. Verifique o caminho e tente novamente.')

else:
    print(f'A pasta Dataset possui {len(arquivos)} arquivos.')

A pasta Dataset possui 7 arquivos.


In [3]:
dfs = {}
def import_df(lista):
    for arquivo in lista:
        chave = arquivo.split('.')[0]
        df = pd.read_csv(f'../Dataset/{arquivo}')
        dfs[chave] = df
    return dfs

try:
    dfs = import_df(arquivos)

except Exception as e:
    print(f'Ocorreu um erro ao importar os arquivos: {e}')

else:
    print(f'Arquivos foram importados e transformados em dataframes\nEles são: {list(dfs.keys())}')

Arquivos foram importados e transformados em dataframes
Eles são: ['studentAssessment', 'vle', 'studentVle', 'assessments', 'studentInfo', 'studentRegistration', 'courses']


Para facilitar a manipulação, e leitura, iremos aplicar aliases nos dataframes.

In [4]:
assessment = dfs['assessments']
courses = dfs['courses']
studentAssessment = dfs['studentAssessment']
studentInfo = dfs['studentInfo']
studentRegistration = dfs['studentRegistration']
studentVle = dfs['studentVle']
vle = dfs['vle']

Passaremos rapidamente por cada df com a intenção de:
- validar a leitura e aliases;
- primeiro contato com df;
- dimensão de tamanhos.

In [17]:
print(f'Tamanho do dataframe: {assessment.shape}\n ')
assessment.head(3)

Tamanho do dataframe: (206, 6)
 


,code_module,code_presentation,id_assessment,assessment_type,date,weight
0,AAA,2013J,1752,TMA,19.0,10.0
1,AAA,2013J,1753,TMA,54.0,20.0
2,AAA,2013J,1754,TMA,117.0,20.0


'Assessment' representa as avaliações de cada curso e suas informações descritivas. 

(206, 6)

In [19]:
print(f'Tamanho do dataframe: {courses.shape}\n ')
courses.head(3)

Tamanho do dataframe: (22, 3)
 


,code_module,code_presentation,module_presentation_length
0,AAA,2013J,268
1,AAA,2014J,269
2,BBB,2013J,268


'Courses' representa o universo de cursos disponíveis na plataforma.

(22, 3)

In [20]:
print(f'Tamanho do dataframe: {studentAssessment.shape}\n ')
studentAssessment.head(3)

Tamanho do dataframe: (173912, 5)
 


,id_assessment,id_student,date_submitted,is_banked,score
0,1752,11391,18,0,78.0
1,1752,28400,22,0,70.0
2,1752,31604,17,0,72.0


'Student Assessment' representa as avaliações realizadas pelos alunos.

(173912, 5)

In [21]:
print(f'Tamanho do dataframe: {studentInfo.shape}\n ')
studentInfo.head(3)

Tamanho do dataframe: (32593, 12)
 


,code_module,code_presentation,id_student,gender,region,highest_education,imd_band,age_band,num_of_prev_attempts,studied_credits,disability,final_result
0,AAA,2013J,11391,M,East Anglian Region,HE Qualification,90-100%,55<=,0,240,N,Pass
1,AAA,2013J,28400,F,Scotland,HE Qualification,20-30%,35-55,0,60,N,Pass
2,AAA,2013J,30268,F,North Western Region,A Level or Equivalent,30-40%,35-55,0,60,Y,Withdrawn


'Student Info' representa as informações descritivas dos alunos inscritos na Open University.

(32593, 12)

In [5]:
print(f'Tamanho do dataframe: {studentRegistration.shape}\n ')
studentRegistration.head(3)

Tamanho do dataframe: (32593, 5)
 


,code_module,code_presentation,id_student,date_registration,date_unregistration
0,AAA,2013J,11391,-159.0,NaN
1,AAA,2013J,28400,-53.0,NaN
2,AAA,2013J,30268,-92.0,12.0


'Student Registration" representa as inscrições de alunos nos cursos.

(32593, 5)

In [16]:
print(f'Tamanho do dataframe: {vle.shape}\n ')
vle.head(3)

Tamanho do dataframe: (6364, 6)
 


,id_site,code_module,code_presentation,activity_type,week_from,week_to
0,546943,AAA,2013J,resource,NaN,NaN
1,546712,AAA,2013J,oucontent,NaN,NaN
2,546998,AAA,2013J,resource,NaN,NaN


'VLE' representa a estrutura pedagógica de atividades dos cursos.

(6364, 6)

### 2.2. Qualidade do Dataset (REFINEMENT)

### 2.3. Distribuições Iniciais (REFINEMENT)

## 3. Data Preparation (BACKLOG)

## 4. EDA (BACKLOG)

### 4.1 Univariada

### 4.2 Bivariada

### 4.3 Multivariada

## 5. Evaluation (BACKLOG)

### 5.1. Insights

### 5.2. Conclusões